pkg/rbac/clusterScopedAuthorization.go

```diff
@@ -77,7 +80,37 @@ func (shared *clusterScopedResources) getClusterScopedResources(cache *Cache, ct
 		}
 	}
 
+	//gather all namespaces in the cluster and cache in shared namespaces cache
+	var allNamespaces []string
+	if len(cache.shared.namespaces) > 0 {
+		klog.V(5).Info("Using namespaces from shared cache")
+		allNamespaces = append(allNamespaces, cache.shared.namespaces...)
+
+	} else {
+
+		klog.V(5).Info("Getting namespaces from Kube Client..")
+
+		cache.resConfig = config.GetClientConfig()
+		clientset, err := kubernetes.NewForConfig(cache.resConfig)
+		if err != nil {
+			klog.Info("Error with creating a new clientset with impersonation config.", err.Error())
```

Comment:
- No impersonation here.
- Log as warning or error.
```suggestion
			klog.Warn("Error with creating a new clientset to get namespaces.", err.Error())
```
```suggestion
			klog.Info("Error with creating a new clientset with impersonation config.", err.Error())
```